src/Rules/Functions/CallToNonExistentFunctionRule.php

```diff
@@ -39,6 +44,14 @@ public function processNode(Node $node, Scope $scope): array
 		}
 
 		if (!$this->reflectionProvider->hasFunction($node->name, $scope)) {
+			// @todo this feels a bit hacky, maybe we need a repository of
+			//    functions that are removed based on version level?
+			if ($this->phpVersion->getVersionId() >= 80000 && (string) $node->name === 'create_function') {
```

Comment:
We already have that: https://github.com/JetBrains/phpstorm-stubs/blob/5cae90f74aa33a4cdbeb4da654525fe665e735c9/Core/Core.php#L755-L768

PHPStan should currently read the @removed annotation That's done here: https://github.com/ondrejmirtes/BetterReflection/blob/348a8777d11b9dd88f13e0d99a84d1671a8cf8a7/src/SourceLocator/SourceStubber/PhpStormStubsSourceStubber.php#L410-L417

But it's probably skipped or unused for functions. It works for classes+methods because RuntimeReflectionProvider is skipped in here: https://github.com/phpstan/phpstan-src/blob/cacc8bba5bc9a755e915bb5433b7e7782c683de6/src/Reflection/ReflectionProvider/ClassBlacklistReflectionProvider.php#L78-L82

But RuntimeReflectionProvider isn't skipped for functions: https://github.com/phpstan/phpstan-src/blob/cacc8bba5bc9a755e915bb5433b7e7782c683de6/src/Reflection/ReflectionProvider/ClassBlacklistReflectionProvider.php#L129-L146

Try to prototype some code in the last location to see if you get anywhere and the function is detected as removed.

You could try to check PhpStormStubsSourceStubber::isPresentFunction() === false which should say that the function no longer exists on PHP 8 and skip in that case.